packages/common/container-definitions/src/loader.ts

```diff
@@ -159,7 +159,7 @@ export interface IContainerEvents extends IEvent {
      *
      * - {@link IContainer.disconnect}
      */
-    (event: "disconnected", listener: () => void);
+    (event: "disconnected", listener: (reason?: string) => void);
```

Comment:
+ @anthony-murphy . This is an interesting problem to discuss (potential precedent for future).
How far should we go in process of exposing observability when it comes to strong contracts? Here, reason is a free-from, can change at any moment. Is this Ok?
The counter example: we map of all SPO errors to internal Fluid errors, and thus guarantee stable API. It's harder here, as errors come from different sources, one of them (socket.io) we do not control, so any bump in dependencies can screw it, and it's very hard to notice (until you hit production).

Thoughts?